include/nanobind/nb_func.h

```diff
@@ -54,22 +60,61 @@ NB_INLINE PyObject *func_create(Func &&func, Return (*)(Args...),
         nargs = sizeof...(Args);
 
     // Determine the number of nb::arg/nb::arg_v annotations
-    constexpr size_t nargs_provided =
+    static constexpr size_t nargs_provided =
```

Comment:
I am curious -- is the addition of `static` for these `constexpr` parameters just general best practice, or do you think it has an actual impact here?

Reply:
This is mostly a habit I have because of past experience with compilers thinking constexpr non-statics need to be captured in lambdas. If I remove the lambda then these definitely aren't necessary. I was confused why some of them were previously static and some not, though.